System.Containers.Tasks/ParseContainerProperties.cs

```diff
@@ -0,0 +1,94 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Resources;
+using System.Text;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using System.Containers;
+
+#nullable disable
+
+namespace System.Containers.Tasks
+{
+    public class ParseContainerProperties : Microsoft.Build.Utilities.Task
+    {
+
+
+        /// <summary>
+        /// The full base image name. mcr.microsoft.com/dotnet/runtime:6.0, for example.
+        /// </summary>
+        [Required]
+        public string FullyQualifiedBaseImageName { get; set; }
+
+        /// <summary>
+        /// The image name for the container to be created.
+        /// </summary>
+        [Required]
+        public string ContainerImageName { get; set; }
+
+        /// <summary>
+        /// The tag for the container to be created.
+        /// </summary>
+        [Required]
+        public string ContainerImageTag { get; set; }
+
+        [Output]
+        public string ParsedContainerRegistry { get; private set; }
+
+        [Output]
+        public string ParsedContainerImage { get; private set; }
+
+        [Output]
+        public string ParsedContainerTag { get; private set; }
+
+        [Output]
+        public string NewContainerImageName { get; private set; }
+
+        [Output]
+        public string NewContainerTag { get; private set; }
+
+        public override bool Execute()
+        {
+            if (!ContainerHelpers.IsValidImageName(ContainerImageName))
+            {
+                Log.LogError("Invalid image name: {0}", ContainerImageName);
```

Comment:
Let's make sure it's hyper clear what param was wrong.

```suggestion
                Log.LogError($"Invalid {nameof(ContainerImageName)}: {0}", ContainerImageName);
```